internal/wantmanager/wantmanager.go

```diff
@@ -17,9 +16,8 @@ var log = logging.Logger("bitswap")
 
 // PeerHandler sends wants / cancels to other peers
 type PeerHandler interface {
-	// Connected is called when a peer connects, with any initial want-haves
-	// that have been broadcast to all peers (as part of session discovery)
-	Connected(p peer.ID, initialWants []cid.Cid)
```

Comment:
This is the big change. By effectively turning the PeerManager into the WantManager (it was already managing _most_ wants anyways), the PeerManager can avoid a bunch of unnecessary work when canceling/broadcasting.